src/pages/index.js

```diff
@@ -54,39 +72,136 @@ const profileEditModal = new ModalWithForm(
   handleProfileEditSubmit
 );
 profileEditModal.setEventListeners();
+
 const addCardModal = new ModalWithForm("#add-card-modal", handleAddCardCreate);
 addCardModal.setEventListeners();
 
+const profileAvatarEditModal = new ModalWithForm(
+  "#profile-avatar-edit-modal",
+  handleProfileAvatarUpdate
+);
+profileAvatarEditModal.setEventListeners();
+
+// Add an event listener to the button to open the avatar update modal
+const profileAvatarEditButton = document.querySelector(
+  "#profile-avatar-edit-button"
+);
+profileAvatarEditButton.addEventListener("click", () => {
+  profileAvatarEditModal.open();
+  profileAvatarInput.value = "";
+});
+
 /*    Form validators    */
-console.log(document.forms);
 const profileEditForm = document.forms["profile-edit-form"];
 const addCardForm = document.forms["add-card-form"];
+const profileAvatarForm = document.forms["profile-avatar-edit-form"];
 const editFormValidator = new FormValidator(config, profileEditForm);
 const addFormValidator = new FormValidator(config, addCardForm);
+const avatarFormValidator = new FormValidator(config, profileAvatarForm);
 
```

Comment:
Please, delete `this.disableButton();` from line 81 in `FormValidator.js`


You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.